### PR TITLE
add wiring for running bare kube-controller-manager

### PR DESCRIPTION
**Pull request by deads2k**
@smarterclayton @liggitt

### CODE REVIEW

File: vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -135,6 +137,8 @@ func NewCMServer() *CMServer {
 
 // AddFlags adds flags for a specific CMServer to the specified FlagSet
 func (s *CMServer) AddFlags(fs *pflag.FlagSet, allControllers []string, disabledByDefaultControllers []string) {
+	fs.StringVar(&s.OpenShiftConfig, "openshift-config", s.OpenShiftConfig, "indicates that this process should be compatible with openshift start master")
```
Comment (smarterclayton) [inline vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/options/options.go]: We might want to make this a hidden flag

Comment (deads2k) [reply vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/options/options.go]: > We might want to make this a hidden flag

Ok.

File: vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch.go [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
+var InformerFactoryOverride informers.SharedInformerFactory
 
-// StartInformers allows overriding inside of the same process.
-var StartInformers func(stop <-chan struct{}) = nil
+func ShimForOpenShift(controllerManager *options.CMServer, clientConfig *rest.Config) (func(), error) {
+	if len(controllerManager.OpenShiftConfig) == 0 {
+		return func() {}, nil
+	}
+
+	// TODO this gets removed when no longer take flags and no longer build a recycler template
```
Comment (deads2k) [inline vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch.go]: I've re-organized a bit and added comments about how we can get rid of each of these patches.

**Comment by deads2k** [comment]
/assign mfojtik
/assign soltysh

I've updated this to be cleaner and have criteria for eliminating the patches we're carrying.  I think its ready to push forward.

**Comment by soltysh** [comment]
David gave me some more details about this PR on irc.
/lgtm

**Comment by deads2k** [comment]
/hold 

I just want to make sure that @smarterclayton is ready to push on this approach. All the options are bad.

**Comment by deads2k** [comment]
This doesn't force us to use it, but does organize our patches to make it clear what we're doing

/hold cancel

**Comment by deads2k** [comment]
@soltysh I added another commit to wire up usage in openshift to avoid duplicate methods.  Take another look?

**Pull request merged**